### PR TITLE
Implement /v1/stat/set/series API

### PR DESCRIPTION
**Pull request by shifucun**
- This is to replace /stat/series/rich API which was implemented for 1 sv * N places, but never used. Below is the prod mixer stats for the past 30 days, there is 0 request to /stat/series/rich
image

- This API takes M sv and N places, return the best ranked stat series for each [sv, place] combination
- This does not take optional argument for filtering. If user is interested in more sources, should use /v1/stat/set/series/all (to be implemented)
- Set version to /v1 
- Sample response: https://github.com/datacommonsorg/mixer/pull/438/files#diff-2e286a38eb97377f678f295dfd4d987abca0302c8fe33122e51166705eb4d8e3
- Once this is in, can migrate browser to use this API, and deprecate /bulk/stats endpoints

### CODE REVIEW

File: internal/server/stat_series.go [hunk 1]
```diff
@@ -288,91 +288,88 @@ func (s *Server) GetStats(ctx context.Context, in *pb.GetStatsRequest) (
...
 		rowList,
 		convertToObsSeriesPb,
 		tokenFn(keyTokens),
 	)
+
 	for token, data := range cacheData {
-		place := strings.Split(token, "^")[0]
+		parts := strings.Split(token, "^")
+		place := parts[0]
+		statVar := parts[1]
```
Comment (pradh) [inline internal/server/stat_series.go:348]: To confirm, what cache is this that has stat-var ID as second part?

File: proto/stat.proto [hunk 2]
```diff
@@ -115,34 +126,22 @@ message GetStatsResponse {
   string payload = 2;
 }
 
-// Request for GetStat API.
+// Request for GetStatSetSeries API.
 // TODO(shifucun): Deprecate GetStats() once all clients are switched to use
 // this API.
```
Comment (pradh) [inline proto/stat.proto:131]: TODO still relevant?

File: proto/stat.proto [hunk 3]
```diff
@@ -58,6 +58,17 @@ message SourceSeries {
...
 
+message Series {
+  // Map from date to stat value.
+  map<string, double> val = 1;
+  // Series metadata.
+  StatMetadata metadata = 2;
+}
+
+message SeriesMap {
+  // Map from key to series. The key could be place dcid or stat var dcid.
```
Comment (pradh) [inline proto/stat.proto:69]: Depending on...?

File: proto/stat.proto [hunk 4]
```diff
@@ -58,6 +58,17 @@ message SourceSeries {
   string provenance_url = 9;
 }
 
+message Series {
```
Comment (pradh) [inline proto/stat.proto:61]: Add comment? Represents a time-series from a source...

**Review by pradh — APPROVED** [review]

File: internal/server/stat_series.go (hunk 1 above)
Comment (shifucun) [reply internal/server/stat_series.go:348]: The place and statvar here is from the request. They are passed into the memcache parallel read so the read-cache can bind back to the original <place, statvar>

File: proto/stat.proto (hunk 4 above)
Comment (shifucun) [reply proto/stat.proto:62]: Done

File: proto/stat.proto (hunk 3 above)
Comment (shifucun) [reply proto/stat.proto]: Updates the comment to make it more generic. What is key is really depending on the response structure so not worth emphasize in here.

File: proto/stat.proto (hunk 2 above)
Comment (shifucun) [reply proto/stat.proto:132]: Yep, website still uses GetStats(), need to do the migration there

**Review by shifucun — COMMENTED** [review]
thanks for review!

**Pull request merged**